lib/bundler/index.rb

```diff
@@ -134,10 +134,16 @@ def size
     def ==(other)
       all? do |spec|
         other_spec = other[spec].first
-        other_spec && (spec.dependencies & other_spec.dependencies).empty? && spec.source == other_spec.source
+        other_spec && dependencies_eql?(spec, other_spec) && spec.source == other_spec.source
       end
     end
 
+    def dependencies_eql?(spec, other_spec)
+      deps       = spec.dependencies.select {|d| d.type != :development }
+      other_deps = other_spec.dependencies.select {|d| d.type != :development }
```

Comment:
Note:  I don't think many things were relying on this equality method since the original `(spec.dependencies & other_spec.dependencies).empty?` seems to be completely wrong.  Indexes are only equal if they share NO dependencies?  I don't even know what that means.